src/socket/socket.js

```diff
@@ -86,19 +84,59 @@ export function initializeSocket(httpServer) {
         return;
       }
       // Example usage of modifyPoints function
-     
+
       const leaderboard = generateLeaderboard(roomId);
       io.to(socket.id).emit("leaderboard", leaderboard);
     });
 
+    let imageUploadOrder = [];
+
+    // Modify the socket.on function to keep track of the image upload order
     socket.on("upload", (data) => {
       console.log("got an image");
       const base64Data = data.image.replace(/^data:image\/\w+;base64,/, "");
       const buffer = Buffer.from(base64Data, "base64");
       const filename = Date.now() + "-" + data.filename;
       const isCorrectObject = checkImage(filename, buffer, "bottle");
       console.log(isCorrectObject);
+
+      // Add the username to the image upload order
+      imageUploadOrder.push(data.username);
+
+      // Call modifyPoints function with the room ID and updated point modifiers
+      modifyPoints(data.roomId);
     });
+    function modifyPoints(roomId) {
+      const room = rooms.find((room) => room.id === roomId);
+      if (!room) {
+        console.log("Room not found");
+        return;
+      }
+
+      // Reset points for all players
+      room.players.forEach((player) => (player.points = 0));
+
+      // Assign points based on the image upload order
+      imageUploadOrder.forEach((username, index) => {
+        const player = room.players.find(
+          (player) => player.username.trim() === username.trim()
+        );
+        if (player) {
+          player.points += index + 1; // Assign points based on the order
```

Comment:
we will use some different approach to assign points to the users, there are some issues with this one (talk to me if you want to know why this isn't the way to do it).

Reply:
ok understand